### ISSUE TITLE
Revert buffer seems to close lsp mode and the server

### ISSUE THREAD

**Issue by ZelCloud** [emacs-lsp/lsp-mode#3469]
### Thank you for the bug report

- [X] You may also try reproduce the issue using clean environment using the following command: `M-x lsp-start-plain`

- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [X] I am using the latest version of `lsp-mode` related packages.

### Bug description

Running the command revert-buffer closes the lsp server.

### Steps to reproduce

Start lsp server in buffer with lsp, then run the command revert-buffer at some point.

### Expected behavior

The lsp server would continue running the lsp-mode won't stop, other minor and major modes don't stop so this behavior seems very odd.

### Which Language Server did you use?

Rust - rust-analyer
Typescript - tls

### OS

Windows

### Anything else?

After a revert-buffer running lsp again starts the server and everything seems to work normally, Doesn't matter how many times this is done.

**Comment by yyoncho** [comment]
I tested with lsp-start-plain (I had to execute additionally `(add-hook 'java-mode-hook 'lsp)`) and it seems like it works fine.

**Comment by ZelCloud** [comment]
I did a bit more debugging, and it seems like certain modes that start lsp on buffer-revert don't restart lsp. So this doesn't seem to be an issue with lsp-mode itself.

I also looked into revert-buffer and found a solution for my issue.

```elisp
;; using this instead of (revert-buffer)
(defun revert-buffer-file-contents ()
  (interactive)
  (let ((active-modes minor-mode-list))
	(revert-buffer t t t) ;;; third t doesnt touch the minor-mode list
	(setq minor-mode-list active-modes)
	)
  )
```

The function above will keep the minor-modes of the buffer but still revert the file, this prevents all the issues I was seeing with the modes restarting and the hooks or minor mode calling additional modes failing.
